### ISSUE TITLE
add a dart fix declarative refactoring for the `whereNotNull()` => `nonNulls` change

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/collection#350]
See https://github.com/dart-lang/collection/pull/332.

I believe this would be a 'replacedBy' dart fix change.

**Comment by oprypin** [comment]
In the past there was a limitation preventing this: can't turn a method call into a getter call. Don't know what the latest status is.

**Comment by devoncarew** [comment]
Yeah, it doesn't appear to be working though it is had to know exactly what syntax to use in the declarative file. I'm guessing that dart fix hasn't seen this exact refactoring before (replacing a deprecated extension method with an extension getter).
